packages/app/src/pages/dashboard/migrate.tsx

```diff
@@ -0,0 +1,314 @@
+	const { t } = useTranslation()
+	const router = useRouter()
+	const dispatch = useAppDispatch()
+	const walletAddress = useAppSelector(selectWallet)
+	const kwentaBalance = useAppSelector(selectKwentaBalance)
+	const claimableBalance = useAppSelector(selectClaimableBalance)
+	const claimableBalanceV2 = useAppSelector(selectClaimableBalanceV2)
+	const stakedKwentaBalance = useAppSelector(selectStakedKwentaBalance)
+	const stakedKwentaBalanceV2 = useAppSelector(selectStakedKwentaBalanceV2)
+	const stakedEscrowedKwentaBalance = useAppSelector(selectStakedEscrowedKwentaBalance)
+	const stakedEscrowedKwentaBalanceV2 = useAppSelector(selectStakedEscrowedKwentaBalanceV2)
+	const totalVestable = useAppSelector(selectTotalVestable)
+	const totalVestableV2 = useAppSelector(selectTotalVestableV2)
+	const kwentaStakingV2Approved = useAppSelector(selectIsKwentaTokenApprovedV2)
+	const kwentaRewards = useAppSelector(selectKwentaRewards)
+
+	const handleGetReward = useCallback(() => {
+		dispatch(getReward())
+	}, [dispatch])
+
+	const tabQuery = useMemo(() => {
+		if (router.query.tab) {
+			const tab = router.query.tab as StakeTab
+			if (Object.values(StakeTab).includes(tab)) {
+				return tab
+			}
+		}
+		return null
+	}, [router])
+
+	const handleUnstakeKwenta = useCallback(
+		() => dispatch(unstakeKwenta(wei(stakedKwentaBalance).toBN())),
+		[dispatch, stakedKwentaBalance]
+	)
+
+	const handleStakeKwentaV2 = useCallback(() => {
+		if (!kwentaStakingV2Approved) {
+			dispatch(approveKwentaToken('kwentaStakingV2'))
+		} else {
+			dispatch(stakeKwentaV2(wei(kwentaBalance).toBN()))
+		}
+	}, [dispatch, kwentaBalance, kwentaStakingV2Approved])
+
+	const [, setCurrentTab] = useState(tabQuery ?? StakeTab.Staking)
+
+	useEffect(() => {
+		if (!!walletAddress) {
+			dispatch(fetchStakingData()).then(() => {
+				dispatch(fetchClaimableRewards())
+				dispatch(fetchStakingV2Data())
+			})
+			dispatch(fetchEscrowData())
+			dispatch(fetchEscrowV2Data())
+		}
+	}, [dispatch, walletAddress])
+
+	const MIGRATE_STEPS = [
+		{
+			key: 'step-1',
+			copy: t('dashboard.stake.tabs.migrate.step-1-copy'),
+			label: t('dashboard.stake.tabs.migrate.rewards'),
+			value: truncateNumbers(claimableBalance, 2),
+			buttonLabel: t('dashboard.stake.tabs.migrate.claim'),
+			onClick: handleGetReward,
+			active: claimableBalance.gt(0),
+		},
+		{
+			key: 'step-2',
+			copy: t('dashboard.stake.tabs.migrate.step-2-copy'),
+			label: t('dashboard.stake.tabs.migrate.staked'),
+			value: truncateNumbers(stakedKwentaBalance, 2),
+			buttonLabel: t('dashboard.stake.tabs.migrate.unstake'),
+			onClick: handleUnstakeKwenta,
+			active: claimableBalance.lte(0) && stakedKwentaBalance.gt(0),
+		},
+		{
+			key: 'step-3',
+			copy: t('dashboard.stake.tabs.migrate.step-3-copy'),
+			label: t('dashboard.stake.tabs.migrate.staked'),
+			value: truncateNumbers(stakedKwentaBalanceV2, 2),
+			buttonLabel: kwentaStakingV2Approved ? t('dashboard.stake.tabs.migrate.stake') : 'Approve',
```

Comment:
We should be consistent with using i18n text, so be good to do this for 'approved' as well.

Reply:
Yes, that's true. Updated.